### PR TITLE
feat: implement `BATCH` support for JSONL

### PR DESCRIPTION
**Pull request by kgpayne**

### CODE REVIEW

**Comment by kgpayne** [comment]
@pnadolny13 this refactors the tests quite a bit to add batch variations, and I see there are upstream changes that need conflicts resolved. Will work on merging upstream tomorrow 🙂👍

File: target_snowflake/connector.py [hunk 1]
```diff
@@ -304,7 +304,7 @@ def _get_file_format_statement(self, file_format):
         return (
             text(
                 f"create or replace file format {file_format}"
-                + "type = 'JSON' compression = 'GZIP'"
+                + "type = 'JSON' compression = 'AUTO'"
```
Comment (pnadolny13) [inline target_snowflake/connector.py:307]: @kgpayne it looks like auto is the default https://docs.snowflake.com/en/sql-reference/sql/create-file-format#type-json. We could optionally exclude it here if you wanted to.

File: tests/batch.py [hunk 2]
```diff
@@ -0,0 +1,182 @@
...
+        current_dir = Path(__file__).resolve().parent
+        return current_dir / "target_test_streams" / f"{self.name}.singer"
+
+
+class SnowflakeTargetBatchArrayData(
+    SnowflakeTargetCustomTestTemplate, SnowflakeTargetArrayData
+):
+    """Test that the target can handle batch messages."""
+
+    name = "batch_array_data"
```
Comment (pnadolny13) [inline tests/batch.py:46]: @kgpayne just to be clear, these arent really asserting anything about the sync other than that it doesnt throw and exception, right? Totally fine for this iteration but in the non-batch default tests I did find some bugs by adding a `validate` assertion, the test jobs didnt throw an exception but the output table wasnt what we'd expect.

**Review by pnadolny13 — APPROVED** [review]
@kgpayne looks good to me! I left a few minor comments but nothing blocking 👍

File: target_snowflake/connector.py (hunk 1 above)
Comment (kgpayne) [reply target_snowflake/connector.py:307]: You are right. I left `compression=` in as a reminder/placeholder for when we come to support other schemes than gzipped JSON 🙂

File: tests/batch.py (hunk 2 above)
Comment (pnadolny13) [reply tests/batch.py:46]: Ah I misread that - perfect!

**Pull request merged**